### PR TITLE
feat(lucky drop): read history from contract

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-4296

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-providers/package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
     "buffer": "^6.0.3",
     "date-fns": "^2.29.3",
     "fortmatic": "^2.2.1",
+    "@ethersproject/abi": "^5.4.0",
```
Comment (guanbinrui) [inline packages/web3-providers/package.json:78]: Use web3 stead if possible.

Comment (guanbinrui) [reply packages/web3-providers/package.json:78]: Remove the original dependence in the RedPacket package.

File: packages/web3-shared/evm/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export * from './types/index.js'
+export * from './types/RedPacket.js'
```
Comment (guanbinrui) [inline packages/web3-shared/evm/src/index.ts]: Move to `@masknet/web3-providers`.

File: packages/web3-providers/package.json (hunk 1 above)
Comment (hanszhoou) [reply packages/web3-providers/package.json:78]: Still some places depend on it in mask package.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**